src/api-extension/cart.service.ts

```diff
@@ -120,8 +151,12 @@ export class CartService {
       anonymousId,
     });
 
-    const deletedCoupons = coupons
-      .filter((coupon) => coupon.status === 'DELETED')
+    const deletedCoupons = coupons.filter(
+      (coupon) => coupon.status === 'DELETED',
+    );
+
+    deletedCoupons
+      .filter((coupon) => coupon.type !== 'promotion_tier')
```

Comment:
Maybe create const for this if it's no defined as a type.

Reply:
I updated Coupon type, but here I don't put constant, because we don't put constants anywhere else in the conditionals/filters